.vib/keycloak/cypress/cypress/integration/keycloak_spec.js

```diff
@@ -4,30 +4,36 @@ import { random } from '../support/utils';
 it('allows creating a new user', () => {
   cy.login();
   cy.get('[href*="/users"]').click();
-  cy.get('#createUser').click();
+  cy.get('button[data-testid="add-user"]').click();
   cy.fixture('users').then((user) => {
-    cy.get('#username').type(`${user.newUser.username}.${random}`);
-    cy.get('#email').type(`${user.newUser.email}.${random}`);
-    cy.get('#firstName').type(`${user.newUser.firstName}.${random}`);
-    cy.get('#lastName').type(`${user.newUser.lastName}.${random}`);
+    cy.get('#kc-username').type(`${user.newUser.username}.${random}`);
+    cy.get('input[data-testid="firstName-input"]').type(`${user.newUser.firstName}.${random}`);
+    cy.get('input[data-testid="lastName-input"]').type(`${user.newUser.lastName}.${random}`);
   });
-  cy.contains('button', 'Save').click();
-  cy.contains('.alert', 'Success');
+  cy.get('button[data-testid="create-user"]').click();
+  cy.contains('h4', 'The user has been created');
 });
 
-it('allows the upload and delete of a locale ', () => {
+it('import and check user information', () => {
   cy.login();
-  cy.contains('Localization').click();
-  cy.contains('Upload localization').click();
-  cy.get('input#import-file').selectFile(
-    'cypress/fixtures/empty-localization-file.json',
-    {
-      force: true,
-    }
-  );
-  cy.fixture('locales').then((locale) => {
-    cy.get('#locale').type(`${locale.German}.${random}`);
+  cy.get('[href*="/realm-settings"]').click();
+  cy.get('div[data-testid="action-dropdown"]').click();
+  cy.get('a[data-testid="openPartialImportModal"]').click();
+  const importFile = 'cypress/fixtures/import-data.json';
+  const randomUsername = `test-realm-user-${random}`;
+  const randomEmail = `test-realm-user-${random}@example.com`;
+  const randomLastname = `user-${random}`;
+  cy.readFile(importFile).then((obj) => {
+    obj.users[0].username = randomUsername;
+    obj.users[0].email = randomEmail;
+    obj.users[0].lastName = randomLastname;
+    const jsonText = JSON.stringify(obj);
+    cy.get('.view-line').type(jsonText, { parseSpecialCharSequences: false, delay: 1 });
```

Comment:
Given that the app gives you to export a JSON file, does it also allow for importing one instead of just typing? If so, you could maybe avoid this logic.

Reply:
It allows importing a file. However, the process implies using a `button` descriptor that Cypress can't use.

![Screen Shot 2022-09-12 at 13 40 52](https://user-images.githubusercontent.com/6660452/189644848-f2ae0c01-78c7-4146-a143-a2778e6996d6.png)